sdks/go/pkg/beam/io/bigqueryio/bigquery.go

```diff
@@ -208,29 +208,60 @@ func mustParseTable(table string) QualifiedTableName {
 // TODO(herohde) 7/14/2017: allow CreateDispositions and WriteDispositions. The default
 // is not quite what the Dataflow examples do.
 
+// WriteOptions represents additional options for executing a write
+type WriteOptions struct {
+	// CreateDisposition specifies the circumstances under which destination table will be created
+	CreateDisposition bigquery.TableCreateDisposition
+}
+
+// newWriteOptions creates a new instance of WriteOptions
+// "CreateIfNeeded" is set as the default write disposition
+func newWriteOptions() WriteOptions {
+	wo := WriteOptions{}
+	wo.CreateDisposition = bigquery.CreateIfNeeded
+	return wo
+}
+
+// WithCreateDisposition specifies the circumstances under which destination table will be created
+func WithCreateDisposition(cd bigquery.TableCreateDisposition) func(wo *WriteOptions) error {
+	return func(wo *WriteOptions) error {
+		wo.CreateDisposition = cd
+		return nil
+	}
+}
+
 // Write writes the elements of the given PCollection<T> to bigquery. T is required
 // to be the schema type.
-func Write(s beam.Scope, project, table string, col beam.PCollection) {
+func Write(s beam.Scope, project, table string, col beam.PCollection, options ...func(*WriteOptions) error) {
 	t := col.Type().Type()
 	mustInferSchema(t)
 	qn := mustParseTable(table)
 
 	s = s.Scope("bigquery.Write")
 
-	// TODO(BEAM-3860) 3/15/2018: use side input instead of GBK.
+	writeOptions := newWriteOptions()
+	for _, opt := range options {
+		if err := opt(&writeOptions); err != nil {
+			panic(err)
+		}
+	}
 
+	// TODO(BEAM-3860) 3/15/2018: use side input instead of GBK.
 	pre := beam.AddFixedKey(s, col)
 	post := beam.GroupByKey(s, pre)
-	beam.ParDo0(s, &writeFn{Project: project, Table: qn, Type: beam.EncodedType{T: t}}, post)
+	beam.ParDo0(s, &writeFn{Project: project, Table: qn, Type: beam.EncodedType{T: t}, Options: writeOptions}, post)
 }
 
+// Add in additional field (CreateDisposition), Bool
 type writeFn struct {
 	// Project is the project
 	Project string `json:"project"`
 	// Table is the qualified table identifier.
 	Table QualifiedTableName `json:"table"`
 	// Type is the encoded schema type.
 	Type beam.EncodedType `json:"type"`
+	// Options specifies additional write options.
+	Options WriteOptions `json:"options"`
```

Comment:
Could potentially be changed to the `writeFn` having the field `CreateDisposition` instead, making it unaware of the `WriteOptions` struct